// @netlify/build/lib/plugins_core/db_setup/validation.js
export const MIGRATION_DIR_PATTERN = /^\d+_[a-z0-9_-]+$/;
export const MIGRATION_FILE_PATTERN = /^\d+_[a-z0-9_-]+\.sql$/;
export const validateMigrations = (dirNames, fileNames, existingSqlFiles) => {
    const errors = [];
    const matchingDirs = [];
    for (const dirName of dirNames) {
        if (!MIGRATION_DIR_PATTERN.test(dirName)) {
            continue;
        }
        matchingDirs.push(dirName);
        if (!existingSqlFiles.has(dirName)) {
            errors.push({ type: 'missing_sql_file', dirName });
        }
    }
    if (errors.length > 0) {
        return { valid: false, errors };
    }
    const matchingFiles = [];
    for (const fileName of fileNames) {
        if (MIGRATION_FILE_PATTERN.test(fileName)) {
            matchingFiles.push(fileName);
        }
    }
    return {
        valid: true,
        dirs: [...matchingDirs].sort(),
        files: [...matchingFiles].sort(),
    };
};
export const formatValidationErrors = (errors) => {
    const lines = errors.map((error) => {
        return `  - "${error.dirName}/migration.sql" is missing.`;
    });
    return `Database migration validation failed:\n${lines.join('\n')}`;
};
