// netlify-cli/dist/commands/database/db-connection.js
import { Client } from 'pg';
import { NetlifyDev } from '@netlify/dev';
import { LocalState } from '@netlify/dev-utils';
import { PgClientExecutor } from './pg-client-executor.js';
export async function connectToDatabase(buildDir) {
    const { client, cleanup } = await connectRawClient(buildDir);
    return {
        executor: new PgClientExecutor(client),
        cleanup,
    };
}
export async function connectRawClient(buildDir) {
    const state = new LocalState(buildDir);
    const storedConnectionString = state.get('dbConnectionString');
    if (storedConnectionString) {
        const client = new Client({ connectionString: storedConnectionString });
        await client.connect();
        return {
            client,
            connectionString: storedConnectionString,
            cleanup: () => client.end(),
        };
    }
    const netlifyDev = new NetlifyDev({
        projectRoot: buildDir,
        aiGateway: { enabled: false },
        blobs: { enabled: false },
        edgeFunctions: { enabled: false },
        environmentVariables: { enabled: false },
        functions: { enabled: false },
        geolocation: { enabled: false },
        headers: { enabled: false },
        images: { enabled: false },
        redirects: { enabled: false },
        staticFiles: { enabled: false },
        serverAddress: null,
    });
    await netlifyDev.start();
    const connectionString = state.get('dbConnectionString');
    if (!connectionString) {
        await netlifyDev.stop();
        throw new Error('Local database failed to start. Set EXPERIMENTAL_NETLIFY_DB_ENABLED=1 to enable.');
    }
    const client = new Client({ connectionString });
    await client.connect();
    return {
        client,
        connectionString,
        cleanup: async () => {
            await client.end();
            await netlifyDev.stop();
        },
    };
}
